src/System.Buffers/src/System/Buffers/DefaultArrayPool.cs

```diff
@@ -84,40 +108,53 @@ public override T[] Rent(int minimumLength)
                 buffer = new T[minimumLength];
             }
 
-            // We had to allocate a buffer, so log that fact.
             if (log.IsEnabled())
             {
-                log.BufferAllocated(
-                    Utilities.GetBufferId(buffer),
-                    buffer.Length,
-                    Utilities.GetPoolId(this),
-                    -1, // no bucket for an on-demand allocated buffer,
-                    index >= _buckets.Length ? 
-                        ArrayPoolEventSource.BufferAllocationReason.OverMaximumSize :
-                        ArrayPoolEventSource.BufferAllocationReason.PoolExhausted);
+                int bufferId = buffer.GetHashCode(), bucketId = -1; // no bucket for an on-demand allocated buffer
+                log.BufferRented(bufferId, buffer.Length, Id, bucketId);
+                log.BufferAllocated(bufferId, buffer.Length, Id, bucketId, index >= _buckets.Length ?
+                    ArrayPoolEventSource.BufferAllocatedReason.OverMaximumSize : ArrayPoolEventSource.BufferAllocatedReason.PoolExhausted);
             }
 
             return buffer;
         }
 
-        public override void Return(T[] buffer, bool clearArray = false)
+        public override void Return(T[] array, bool clearArray = false)
         {
-            if (buffer == null)
-                throw new ArgumentNullException(nameof(buffer));
+            if (array == null)
+            {
+                throw new ArgumentNullException(nameof(array));
+            }
+            else if (array.Length == 0)
+            {
+                // Throw away empty arrays.
```

Comment:
Is it really throwing it away? I thought that a rented 0 length array would be static. 

Though i guess you could return a different 0-length array  (i.e. one that wasn't rented) and we still wouldn't need to do anything for it. So this is fine.